datafusion/tests/sql.rs

```diff
@@ -1754,51 +1779,70 @@ async fn equijoin_and_unsupported_condition() -> Result<()> {
 #[tokio::test]
 async fn left_join() -> Result<()> {
     let mut ctx = create_join_context("t1_id", "t2_id")?;
-    let sql = "SELECT t1_id, t1_name, t2_name FROM t1 LEFT JOIN t2 ON t1_id = t2_id ORDER BY t1_id";
-    let actual = execute(&mut ctx, sql).await;
+    let equivalent_sql = [
```

Comment:
I wonder if this test case and those below it add any additional coverage compared to 

```rust
async fn equijoin_multiple_condition_ordering() -> Result<()> {
```

Like are there bugs that would cause `equijoin_multiple_condition_ordering` to pass but the others to fail?

Reply:
My main objective was to ensure nothing specific to equijoin (`INNER`) would break this implementation for the other join types (which occurred when developing it). This may be more relevant if any further optimisations occur.